pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1307,9 +1412,25 @@ func (nc *Controller) cancelPodEviction(node *v1.Node) bool {
 
 // evictPods queues an eviction for the provided node name, and returns false if the node is already
 // queued for eviction.
-func (nc *Controller) evictPods(node *v1.Node) bool {
+// If node was already evicted, the pods are deleted immediately.
+func (nc *Controller) evictPods(node *v1.Node, pods []*v1.Pod) bool {
 	nc.evictorLock.Lock()
 	defer nc.evictorLock.Unlock()
+	if nc.nodeEvictionMap.getStatus(node.Name) == evicted {
```

Comment:
With this change, evictorLock is protecting a section that may be potentially extremely long (DeletePods may take seconds).
I think we need to change that.

Reply:
Goroutine added.